parser/src/aspects/call.rs

```diff
@@ -113,20 +114,40 @@ impl<'a> CallAspect<'a> for Parser<'a> {
         }))
     }
 
-    fn expand_call_chain(&mut self, mut expr: Expr<'a>) -> ParseResult<Expr<'a>> {
+    fn expand_member_chain(&mut self, mut expr: Expr<'a>) -> ParseResult<Expr<'a>> {
         while self
             .cursor
             .lookahead(
                 of_types(&[TokenType::RoundedLeftBracket, TokenType::SquaredLeftBracket])
-                    .or(blanks().then(of_type(TokenType::Dot).and_then(identifier_parenthesis()))),
+                    .or(blanks()
+                        .then(of_type(TokenType::Dot).and_then(of_type(TokenType::Identifier)))),
             )
             .is_some()
         {
             self.cursor.advance(blanks());
             if self.cursor.peek().token_type == TokenType::SquaredLeftBracket {
                 expr = self.parse_subscript(expr).map(Expr::Subscript)?;
-            } else {
+            } else if self
+                .cursor
+                .lookahead(
+                    of_type(TokenType::Dot)
+                        .and_then(identifier_parenthesis())
+                        .or(of_types(&[
+                            TokenType::RoundedLeftBracket,
+                            TokenType::SquaredLeftBracket,
+                        ])),
+                )
```

Comment:
This code is very similar to the one at the start of the method. Maybe start by parsing a field access, then see if there are parenthesis/square brackets.